### PR TITLE
WIP: Add csv renderer

### PR DESCRIPTION
**Pull request by colanconnon**

### CODE REVIEW

File: webgrid/flask.py [hunk 1]
```diff
@@ -50,3 +50,8 @@ def xls_as_response(self, workbook, filename):
         buf.seek(0)
         return send_file(buf, mimetype='application/vnd.ms-excel',
                          as_attachment=True, attachment_filename=filename)
+
+    def csv_as_response(self, csv, filename):
+        csv.seek(0)
+        return send_file(csv, mimetype='application/vnd.ms-excel',
```
Comment (tjlevel12) [inline webgrid/flask.py]: Any reason not to use `text/csv` for the mimetype?

File: webgrid/tests/test_rendering.py [hunk 2]
```diff
@@ -2,11 +2,16 @@
 
 import datetime as dt
 from io import BytesIO
+try:
```
Comment (tjlevel12) [inline webgrid/tests/test_rendering.py]: As above, `from six import StringIO`.

File: webgrid/flask.py (hunk 1 above)
Comment (colanconnon) [reply webgrid/flask.py]: Nope. I will switch it

File: webgrid/templates/grid_footer.html [hunk 3]
```diff
@@ -2,6 +2,10 @@
     {% if not grid.hide_excel_link %}
         <p><a class="export-link" href="{{ renderer.export_url() }}">export to Excel</a></p>
     {% endif %}
+    
+    {% if not grid.hide_csv_link %}
```
Comment (colanconnon) [inline webgrid/templates/grid_footer.html]: @guruofgentoo not sure if we want a flag for showing the csv link or if we want to make the excel link more dynamic and manipulate `default_spreadsheet_format` to control how we are exporting from the grid

Comment (guruofgentoo) [reply webgrid/templates/grid_footer.html]: I would rather have a single flag to control show/hide for export links in general, and then show links based on the allowed renderers for the grid.

**Comment by guruofgentoo** [comment]
I don't like how `allowed_export_targets` relates to the renderers initialized in `set_renderers`. I understand that was there from the previous PR, but we need to change it:
- `allowed_export_targets` should be a dict of string keys to renderer classes
  - default value is simply the Excel renderer, whether that is xls or xlsx depends on the lib installed (as it is elsewhere)
- change `set_export_to` to check the input against the dict keys
- change `set_renderers` to set an attribute on the grid named by the string key, with a value of the initialized renderer
  - HTML can stay as-is for now
- the HTML template should have its footer export links area controlled via a single flag
  - `hide_excel_link` will also control this for now, but be deprecated
  - links should include anything defined in `allowed_export_targets`

File: webgrid/renderers.py [hunk 4]
```diff
@@ -847,3 +849,50 @@ def as_response(self, wb=None, sheet_name=None):
             wb.close()
         wb.filename.seek(0)
         return self.grid.manager.file_as_response(wb.filename, self.file_name(), self.mime_type)
+        return self.grid.manager.xls_as_response(wb, self.file_name())
```
Comment (guruofgentoo) [inline webgrid/renderers.py]: Guessing you don't want the second return here.

**Comment by guruofgentoo** [comment]
One more thing - let's arrange the export links more like this:

untitled

**Comment by colanconnon** [comment]
@guruofgentoo should be good to go now.

**Pull request merged**